spec/lib/active_versioning/model/versioned_spec.rb

```diff
@@ -0,0 +1,122 @@
+require 'spec_helper'
+
+RSpec.describe ActiveVersioning::Model::Versioned do
+  let!(:user) { ActiveVersioning::Test::User.create(name: 'Bo Janglez', email: 'bo.janglez@gmail.com') }
+
+  subject { ActiveVersioning::Test::Post.create(title: 'So Post', body: 'Such interesting.  Very wow.', user: user) }
+
+  it { should have_many(:versions) }
+
+  describe "after create" do
+    it "creates a new version" do
+      expect {
+        ActiveVersioning::Test::Post.create(title: 'So Post', body: 'Such interesting.  Very wow.', user: user, version_author: user)
+      }.to change { ActiveVersioning::Test::Version.count }.by 1
+    end
+  end
+
+  describe "#version_author" do
+    it "should be able to set and view the author" do
+      subject.version_author = user
+
+      expect(subject.version_author).to eq user
+    end
+  end
+
+  describe "#live?" do
+    it { expect(subject.live?).to eq true }
+  end
+
+  describe "#version?" do
+    it { expect(subject.version?).to eq false }
+  end
+
+  describe "#current_draft" do
+    context "when the record is not persisted" do
+      subject { ActiveVersioning::Test::Post.new }
+
+      it "raises an error" do
+        expect { subject.current_draft }.to raise_error(ActiveVersioning::Model::Versioned::RecordNotPersisted)
+      end
+    end
+
+    context "when there are no draft versions" do
+      it "creates a new draft version wrapped in a VersionProxy" do
+        expect {
+          subject.current_draft
+
+          # Verify we have a VersionProxy
+          expect(subject.current_draft.__getobj__).to eq subject
+        }.to change { subject.versions.count }.by 1
```

Comment:
would like to see this split into two tests, one for the new draft version, one testing the draft's `__getobj__` is the proper object